internal/pkg/cli/env_delete.go

```diff
@@ -348,6 +372,80 @@ func (o *deleteEnvOpts) ensureRolesAreRetained() error {
 	return nil
 }
 
+// emptyBuckets returns nil if buckets were deleted successfully. Otherwise, returns the error.
+func (o *deleteEnvOpts) emptyBuckets() error {
+	envStack, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: []*string{aws.String("cloudformation")},
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("find env stack resource: %w", err)
+	}
+	if len(envStack.ResourceTagMappingList) > 1 {
+		return fmt.Errorf("ambiguous env stacks found")
+	} else if len(envStack.ResourceTagMappingList) < 1 {
+		return fmt.Errorf("no env stack found")
+	}
+	stackID := envStack.ResourceTagMappingList[0].ResourceARN
+
+	s3buckets, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: []*string{aws.String("s3:bucket")},
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(envS3BucketStackNameTagKey),
+				Values: []*string{aws.String(stack.NameForEnv(o.appName, o.name))},
+			},
+			{
+				Key:    aws.String(envS3BucketStackIDTagKey),
+				Values: []*string{stackID},
+			},
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+
+	if err != nil {
+		return fmt.Errorf("find s3 bucket resources: %w", err)
+	}
+
+	var failedBuckets []string
+	emptyBucketFailed := false
```

Comment:
nit: `emptyBucketFailed` gets initialized to the zero value for `bool`, which is `false`
```suggestion
	var emptyBucketFailed bool
```